packages/react-next/src/Utilities/Customizer.test.tsx

```diff
@@ -0,0 +1,280 @@
+      wrapper => {
+        expect(wrapper.html()).toEqual('<div>field2</div>');
+      },
+    );
+  });
+
+  it('can use a function to merge settings', () => {
+    safeMount(
+      <Customizer settings={{ field: 'field1' }}>
+        <Customizer
+          settings={
+            // tslint:disable-next-line:jsx-no-lambda
+            (settings: { field: string }) => ({ field: settings.field + 'field2' })
+          }
+        >
+          <Bar />
+        </Customizer>
+      </Customizer>,
+      wrapper => {
+        expect(wrapper.html()).toEqual('<div>field1field2</div>');
+      },
+    );
+  });
+
+  it('can suppress updates', () => {
+    Customizations.applySettings({ field: 'globalName' });
+
+    safeMount(
+      <Customizer settings={{ nonMatch: 'customName' }}>
+        <Bar />
+      </Customizer>,
+      wrapper => {
+        // verify base state
+        expect(wrapper.html()).toEqual('<div>globalName</div>');
+
+        ReactTestUtils.act(() => {
+          // verify it doesn't update during suppressUpdates(), and it works through errors, and it updates after
+          Customizations.applyBatchedUpdates(() => {
+            Customizations.applySettings({ field: 'notGlobalName' });
+            // it should not update inside
+            expect(wrapper.html()).toEqual('<div>globalName</div>');
+            throw new Error();
+          });
+        });
+        // afterwards it should have updated
+        expect(wrapper.html()).toEqual('<div>notGlobalName</div>');
+
+        ReactTestUtils.act(() => {
+          // verify it doesn't update during suppressUpdates(), works through errors, and can suppress final update
+          Customizations.applyBatchedUpdates(() => {
+            Customizations.applySettings({ field: 'notUpdated' });
+            // it should not update inside
+            expect(wrapper.html()).toEqual('<div>notGlobalName</div>');
+            throw new Error();
+          }, true);
+        });
+
+        // afterwards, it should still be on the old value
+        expect(wrapper.html()).toEqual('<div>notGlobalName</div>');
+
+        ReactTestUtils.act(() => {
+          // verify it updates after suppressUpdates()
+          Customizations.applySettings({ field2: 'lastGlobalName' });
+        });
+
+        expect(wrapper.html()).toEqual('<div>notUpdatedlastGlobalName</div>');
+      },
+    );
+  });
+});
+
+describe('Customizer wraps ThemeProvider', () => {
+  it('provide theme using ThemeProvider', () => {
+    safeMount(
+      <Customizer settings={{ theme: { fonts: { medium: { fontWeight: 800 } } } }}>
+        <div>Content</div>
+      </Customizer>,
+      wrapper => {
+        expect(wrapper.html()).toEqual('<div><div>Content</div></div>');
+        const inlineStyle = wrapper.find('div').get(0).props.style;
+        expect(inlineStyle).toBeDefined();
+        expect(Object.keys(inlineStyle).length).toBeGreaterThan(0);
+      },
+    );
+  });
+
+  fit('disable ThemeProvider', () => {
```

Comment:
fit?

Reply:
ugh, thanks for catching! i wish there is a way to warn me on using `fit`